src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -230,6 +230,29 @@ def _validate_non_args(self):
                         "yet has an ArgumentInterface interface."
                         "".format(str(symbol)))
 
+    def get_unlinked_datasymbols(self, ignore_precision=True):
```

Comment:
I think "unlinked" should be "deferred" as this method could be called at any time. It also returns the names not the symbols so the name should reflect this. I also suggest ignore_precision defaults to False. You also use "unlinked", "deferred", "unresolved" and "unknown" in various places. I would suggest "deferred" here and "unresolved" as a variable name if all of the symbols have been added to the symbol table.

Reply:
I've gone for 'get_unresolved_datasymbols' and switched to using unresolved throughout.